connector.go

```diff
@@ -25,6 +25,8 @@ type AuthConfig struct {
 
 	ClientCert []byte
 	ClientKey  []byte
+
+	Headers map[string][]string
```

Comment:
If I am trying to place the headers somewhere like queryparams, it would be a part of https call or payload.
Instead, if we are placing the AuthConfig, it is easy to us headers from the .go files, then easy to add it to the request Header in Build Request.

Reply:
It would be helpful to have an example where we make calls with different headers. Could you please update this PR with an example.
One call with `x-header: 1` and another call with `x-header: 2`
@hemanthKa677

Reply:
There would be cases where same connection can be used to hit different NIOS hosts which would have different OPHID headers